src/model/model_handler.py

```diff
@@ -75,34 +72,30 @@ def load_model(self, model_path: str) -> None:
 
     def train(self, train_loader: torch.utils.data.DataLoader):
         self.model.train()
-    
+
+        batch_results_list: list[Results] = []
+
         for batch in train_loader:
-            train_weights = torch.ones_like(batch["operator"].y)
-            train_weights[batch["operator"].y == 0] = self.neg_weight
-            train_weights[batch["operator"].y == 1] = self.pos_weight
             self.optimizer.zero_grad()
             out = self.model(batch.x_dict, batch.edge_index_dict)
 
             # Torch eval requires one dimensional tensors and these are X,1
             metric_result = self.eval_metric(out["operator"].squeeze(), batch["operator"].y.squeeze())
-
-            loss = self.loss_function(
-                out["operator"], batch["operator"].y, weight=train_weights
+            results = metrics.compute_results(
+                batch, self.model, self.pos_weight, self.neg_weight, self.loss_function, self.eval_metric
             )
-            loss.backward()
+            batch_results_list.append(results)
+            results.loss.backward()
             self.optimizer.step()
-        # loss, pred, original
-        return Results(loss, metric_result, out["operator"], batch["operator"].y)
-
-    def get_action_predictions(model: torch.nn.Module, state: torch.Tensor) -> torch.Tensor:
-        pass
+        
+        epoch_mean_results = Results.reduce_list_of_results(batch_results_list)
+        return epoch_mean_results
```

Comment:
Previously we were reporting the results of last batch as epoch results, now for each epoch we get the average of all metrics. The `Results.reduce_lists_of_results` is a convenience method to get the average of all results